compiler/src/iree/compiler/Codegen/LLVMGPU/Utils/LLVMGPULayoutAnalysisAndDistribution.cpp

```diff
@@ -0,0 +1,485 @@
+                     {DimType::VecIdY, 2}}};
+          return {{{DimType::LaneIdY, 8},
+                   {DimType::LaneIdZ, 1},
+                   {DimType::VecIdZ, 1}}};
+        case MMAMatrixType::CMatrix:
+          if (dim == 0)
+            return {{{DimType::LaneIdY, 8},
+                     {DimType::VecIdY, 2},
+                     {DimType::LaneIdZ, 1}}};
+          return {{{DimType::VecIdX, 2},
+                   {DimType::LaneIdX, 4},
+                   {DimType::VecIdZ, 1}}};
+      }
+      break;
+    default:
+      return {};
+  }
+}
+
+static std::array<int, 2> getMMACanonicalShape(MMAType mmaType,
+                                               MMAMatrixType matrixType) {
+  switch (mmaType) {
+    case MMAType::M16N8K16:
+      switch (matrixType) {
+        case MMAMatrixType::AMatrix:
+          return {16, 16};
+        case MMAMatrixType::BMatrix:
+        case MMAMatrixType::CMatrix:
+          return {16, 8};
+      }
+      break;
+    default:
+      return {};
+  }
+}
+
+Layout::Layout(const DimOrderArray &dimOrder,
+               const std::array<int, maxTensorDims> &canonicalShape) {
+  assert((dimOrder.size() > 0) && (dimOrder.size() <= maxTensorDims));
+  for (int i = 0; i < maxTensorDims; i++) {
+    int j;
+    for (j = 0; j < dimOrder[i].size(); j++) {
+      Dimension dim = dimOrder[i][j];
+      order[i][j] = dim.type;
+      shape[dim.type] = dim.value;
+    }
+    // Add batch dimension to the end
+    if (i == 0) {
+      order[i][j] = DimType::Batch0;
+      shape[DimType::Batch0] = 1;
+    } else {
+      order[i][j] = DimType::Batch1;
+      shape[DimType::Batch1] = 1;
+    }
+    this->canonicalShape[i] = canonicalShape[i];
+  }
+  rank = dimOrder.size();
+}
+
+void Layout::reduceDim(int i) {
+  assert(rank > 0);
+  assert(order.size() > i);
+  for (auto dim : order[i]) {
+    shape[dim] = 1;
+  }
+  rank--;
+}
+
+void Layout::updateBatchDims(int dim0, int dim1) {
+  shape[DimType::Batch0] = dim0 / canonicalShape[0];
+  shape[DimType::Batch1] = dim1 / canonicalShape[1];
+}
+
+AffineExpr Layout::computeDim(int i, const DimArray &state,
+                              OpBuilder &builder) {
+  AffineExpr d0, d1, d2;
+  bindDims(builder.getContext(), d0, d1, d2);
+  AffineExpr dim = builder.getAffineConstantExpr(0);
+  AffineExpr dimScale = builder.getAffineConstantExpr(1.0);
+  for (const auto &dimType : order[i]) {
+    switch (dimType) {
+      case DimType::LaneIdX:
+        dim = dim + dimScale * d0;
+        break;
+      case DimType::LaneIdY:
+        dim = dim + dimScale * d1;
+        break;
+      case DimType::LaneIdZ:
+        dim = dim + dimScale * d2;
+        break;
+      default:
+        dim = dim + dimScale * builder.getAffineConstantExpr(state[dimType]);
+        break;
+    }
+    dimScale = dimScale * builder.getAffineConstantExpr(shape[dimType]);
+  }
+  return dim;
+}
+
+std::string Layout::str() const {
+  std::stringstream ss;
```

Comment:
can you convert that to LLVM style of printing out debug info.
Example:
https://github.com/openxla/iree/blob/624db6dc2be50b64c519353bdbc8c5dc8dce3280/compiler/src/iree/compiler/Codegen/Common/GPUDistributeSharedMemoryCopy.cpp#L37

Reply:
sure will change.